### ISSUE TITLE
Feature Request: Atom Editor Integration

### ISSUE THREAD

**Issue by rriemann** [google/git-appraise#85]
Hello,

I open this topic just to start a joint discussion on what it takes to offer a GUI integration for the popular Atom Editor.

I also asked in the Atom Editor Forum to find like-minded persons:

https://discuss.atom.io/t/gui-for-git-appraise-code-review-comments/52261

**Comment by ojarjur** [comment]
Looking at that link, your main goal is to support comments in a text editor.

That is exactly one of the use cases the git-appraise format is meant to support. That is why comments are stored in a different ref from everything else; so that a tool that only cares about comments can use just those.

There is one unimplemented feature that you may want in an editor, though; #67

The notes format that we use represents comment actions (e.g. "add a comment", "reply to a comment", etc). However, there is no action yet implemented for editing a comment. It would not be hard to add such a thing, but it hasn't been done yet.

On a separate note; in the linked thread there was a concern raised about whether git-appraise comments are "transitory" or are meant for long-term storage. To answer that; git-appraise comments are not transitory... they are meant to last indefinitely. For instance, the comments in a code review should never go away.

**Comment by ojarjur** [comment]
@rriemann Yes; comments have a start line, start column, end line, and end column.

**Comment by rriemann** [comment]
So the features #67  has been implemented in the meanwhile. :tada: 

I don't estimate that I can resolve this request my own. I'm not familar with atom plugins implementation.

Do you have tips what can be undertaken to make this happen?

I need an integration to demonstrate with a demonstrator how GDPR compliant policy writing is possible for the public sector. My preparation so far: https://riemann.cc/files/share/cast-writing/slides-cast-writing-edps.html

**Comment by ojarjur** [comment]
I also do not have any familiarity with atom plugins, but will leave this thread open in case someone else wants to chime in.

**Comment by mcansky** [comment]
any news on this ?

**Comment by ojarjur** [comment]
@mcansky As far as I am aware, this is still waiting for someone to take it on.

**Comment by mcansky** [comment]
@ojarjur : any suggestion as to where to start beyond the plugin tutorial to get something started? I was wondering if starting from an existing Git plugin could prove helpful

**Comment by ojarjur** [comment]
@mcansky Sorry, I do not have any experience with the Atom editor, so I can't provide any guidance on that.

However, I would say that the https://github.com/google/git-appraise/issues/97 might be applicable to your effort, if you want to have comments outside the context of a code review. I have implemented that feature in a branch, but not yet created the pull request to merge it in yet (it's blocked on an earlier pull request that is still pending).

**Comment by mcansky** [comment]
thanks
